[Abhishek Gupta] iP

Agentum07 opened the pull request:
# Bhutu Pro
> "If you take it one task at a time, you can do anything you set your mind to." - Bhutu

The Bhutu ChatBot frees your mind of having to remember things you need to do. It's,
* Text-based
* Easy to learn
* ~~FAST~~ _SUPER_ FAST to use!

All you need to do is - 
1. Download the code from [here](https://github.com/Agentum07/ip)
2. Double click it
3. Add your tasks
4. Let it handle the rest :grin:

**And best of all, it is FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders *(coming soon!)*

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        new Duke().run();
    }
}
```

Commit:
Add Gradle support

Agentum07 committed:
Feat: Greet, Echo, Exit

Agentum07 committed:
Feat: Add, List

Agentum07 committed:
Feat: Mark as done

Agentum07 committed:
Feat: Todo, Event, Deadline

Agentum07 committed:
Feat: Automated UI testing

Agentum07 committed:
Feat: error handling

Agentum07 committed:
Feat: Delete functionality

Agentum07 committed:
Feat: Save functionality

Agentum07 committed:
Feat: Dates and times

Agentum07 committed:
Merge branch 'branch-Level-8'

Agentum07 committed:
Add more OOP

Agentum07 committed:
Add to package, duke and duke.task

Agentum07 committed:
Feat: JUnit testing
Update level8

Agentum07 committed:
Tweak JavaDoc

Agentum07 committed:
Fix JavaDoc

Agentum07 committed:
A-CodingStandard

Agentum07 committed:
Merge branch 'branch-A-CodingStandard'

Agentum07 committed:
Merge branch 'branch-A-JavaDoc'

Agentum07 committed:
Feat: find a

Agentum07 committed:
Merge branch 'branch-level-9'

Agentum07 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

Agentum07 committed:
Fix: comply to java coding standards

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
```
Perhaps need a descriptive header comments for the Duke class? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+    }
+
+    /**
+     * entry point of the duke chatbot.
```
Perhaps need an empty line between description and parameter section? 🤔 
Perhaps the first word need to start with uppercase? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+    }
+
+    /**
+     * interacts with the user
```
Perhaps need an empty line between description and parameter section? 🤔 
Perhaps the first word need to start with uppercase? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+    }
+
+    /**
+     * The main function of Bhutu
```
Perhaps need an empty line between description and parameter section? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+    /**
+     * entry point of the duke chatbot.
+     * @throws DukeException in case any unexpected input is passed
+     * @throws IOException in case of file issues
```
Perhaps need a full stop at the end? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+    /**
+     * interacts with the user
+     * @throws DukeException if input is wrong
+     * @throws IOException thrown if file handling fails
```
Perhaps need a full stop at the end? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+
+        Scanner sc = new Scanner(System.in);
+        boolean flag = true;
+        while(flag) {
```
Perhaps need a space between `while` and `(`? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+                switch (command) {
+                case "list":
+                    output = items.printList();
+
```
Perhaps need to remove the empty line? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,15 @@
+package duke;
+
+/**
+ * Exception for the Duke app
```
Perhaps need a full stop at the end? 🤔

HolmesJJ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,154 @@
+
+    /**
+     * The main function of Bhutu
+     * @param args The command line arguments
```
Perhaps need a full stop at the end? 🤔

HolmesJJ commented on src/main/java/duke/Items.java:
```diff
@@ -0,0 +1,135 @@
+
+/**
+ * The items in the bot
+ * responsible for adding things to the list of items
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,116 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
```
Perhaps need a descriptive header comments for the Parser class? 🤔

HolmesJJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+                boolean isDone = data[1].equals("1");
+                Task task = null;
+                switch (taskType) {
+                    case "T":
```
Perhaps need to remove the indentation for case clauses? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
```
Perhaps need a descriptive header comments for the Parser class? 🤔

HolmesJJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+    private ArrayList<String> fileTasks;
+
+    /**
+     * Constructor
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+     * @return The String of the queried line
+     * @throws DukeException When the line is not found
+     */
+    public String getFileLine(int index) throws DukeException{
```
Perhaps need a space between `DukeException` and `{` ? 🤔

HolmesJJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+     */
+    private static final String SPACE = "    ";
+    private static final String LINE = SPACE + "______________________________________________________________________";
+    private static final String END_LINE = SPACE + "======================================================================\n";
```
Should this be wrapped to next line? 🤔

HolmesJJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+
+import java.util.Scanner;
+
+public class Ui {
```
Perhaps need a descriptive header comments for the Parser class? 🤔

HolmesJJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * returns a greeting to the user
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Gets the user input
```
Perhaps need an empty line between description and parameter section? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+package duke.task;
+
+/**
+ * A task of type deadline
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,27 @@
+package duke.task;
+
+/**
+ * A task of type Event
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,50 @@
+package duke.task;
+
+/**
+ * the task in the list
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,10 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DukeTest {
```
Perhaps need an empty line between description and parameter section? 🤔

HolmesJJ commented on src/test/java/duke/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TaskTest {
```
Perhaps need an empty line between description and parameter section? 🤔

HolmesJJ commented on src/test/java/duke/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+        Items item = new Items();
+        String msg;
+        msg = item.addItem(new Todo("abc"));
+        assertEquals("Got it, I've added this task:\n" +
```
Should the `+` be at next line ? 🤔

HolmesJJ commented on src/test/java/duke/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+        } catch (DukeException e) {
+            msg = "";
+        }
+        assertEquals("Nice! I've marked this task as done: \n" +
```
Should the `+` be at next line ? 🤔

HolmesJJ commented on src/test/java/duke/TaskTest.java:
```diff
@@ -0,0 +1,46 @@
+        } catch (DukeException e) {
+            msg = "";
+        }
+        assertEquals("These are your tasks: \n" +
```
Should the `+` be at next line ? 🤔

HolmesJJ commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,24 @@
+package duke.task;
+
+/**
+ * A task of type todo
```
Perhaps need a full stop at the end? 🤔
And I noticed the same issue in several other places below.

HolmesJJ commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+        try {
+            return fileTasks.get(index - 1);
+        } catch (Exception e) {
+            throw new DukeException("Task doesn't exist" +
```
Should the + be at next line ? 🤔

HolmesJJ reviewed (commented):
LGTM. Very clear code and easy to understand. 😃 Just some small code standard issues need to fix.

Agentum07 committed:
Fix: comply to java coding standards

MGRL2201 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,116 @@
+
+public class Parser {
+    /**
+     * combines an array of strings into a space seperated sentence.
```
Perhaps capitalize the C?

MGRL2201 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * reads file. If file does not exist, create file
```
Perhaps capitalize the R and add a full stop?

MGRL2201 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * updates a line in the file at the given index
```
Perhaps you can capitalize the U, add a full stop, and have an empty line between your header and param.

MGRL2201 reviewed (commented):
Your code is easy to read and elaborate. Just some minor issues with your javadocs like the capitalization, missing empty lines, and full stops. Well done!

Chesterwongz commented on src/main/java/duke/Items.java:
```diff
@@ -13,35 +13,35 @@
      * Constructor for Items
      * @param tasks An ArrayList of Tasks
      */
     public  Items(ArrayList<Task> tasks) {
```
seems like an extra space here

Chesterwongz reviewed (commented):
LGTM, other than some minor styling issues. With Checkstyle, this should all go away. Great work!

Agentum07 committed:
Fix: comply with Java coding standards

Agentum07 committed:
Feat: A-Gradle

Agentum07 committed:
Fix: comply to java coding standards

Agentum07 committed:
Merge branch 'branch-A-Gradle'

Agentum07 committed:
Fix: A-Checkstyle

Agentum07 committed:
Merge branch 'branch-A-Checkstyle'

Agentum07 committed:
Fix: add duke.java to package

Agentum07 committed:
Fix: optimize code structure

Agentum07 committed:
Feat: Add GUI

Agentum07 committed:
Merge branch 'branch-Level-10'

Agentum07 committed:
Feat: Add varags

Agentum07 committed:
Feat: A-Assetions

Agentum07 committed:
Code has underlying assumptions which if not satisfied, can crash the
program during execution.

Add Assertions at key points to verify that the assumptions made
are true.

Assertions are being adopted because they help define assumptions about
the program state so the runtime can verify them.

Agentum07 committed:
Feat: A-Assertions

Code has some underlying assumptions that if unsatisfied, can crash the program.

Add assertions will check those assumptions at key points in the code and if unsatisfied, exit the program.

Adopted assertions because they help define assumptions about the program state so the runtime can verify them.

Agentum07 committed:
Merge branch 'branch-A-Assertions'

Agentum07 committed:
Merge branch 'master' of https://github.com/Agentum07/ip

Agentum07 committed:
Chore: Updated Bot image

Agentum07 committed:
Chore: fix coding standards

Agentum07 committed:
Chore: A-CodingStandard

Coding Standard: make code follow the coding standard.

Code currently does not follow the coding standard.

A coding standard following code improves readability and more programmer
friendly.

Let's
* make sure the code is upto standard

Agentum07 committed:
Merge branch 'master' into branch-A-CodingStandard

Agentum07 committed:
Merge branch 'branch-A-CodingStandard'

Agentum07 committed:
Merge pull request #2 from Agentum07/branch-A-CodingStandard

Chore: fix coding standards

Agentum07 committed:
Merge branch 'branch-A-CodingStandard' of https://github.com/Agentum07/ip

Agentum07 committed:
Merge branch 'master' of https://github.com/Agentum07/ip

* 'master' of https://github.com/Agentum07/ip:

Agentum07 committed:
Shift response functionality to UI

Agentum07 committed:
Erroneous commit, revert back 2 commits prior

Agentum07 committed:
Revert to pre-shift functionality

Agentum07 committed:
Revert to pre-shift functionality

Agentum07 committed:
Feat: Add ability to undo last task

Agentum07 committed:
Improve GUI fitting

Agentum07 committed:
Merge branch 'master' into branch-C-Undo

Agentum07 committed:
Merge branch undo with master branch

Merge Branch-C-Undo

Agentum07 committed:
Merge branch 'branch-C-Undo'

Agentum07 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Duke.java

Agentum07 committed:
fix merge conflicts

Agentum07 committed:
Chore: fix bot output

Agentum07 committed:
Fix code quality

Agentum07 committed:
Chore: fix bot output

Agentum07 committed:
Fix code structure

Agentum07 committed:
Merge branch 'branch-C-Undo'. Update undo functionality

Agentum07 committed:
Feat: Add Help command.

Agentum07 committed:
Feat: Add Help functionality

Agentum07 committed:
Chore: refactor to better fit code standards

Agentum07 committed:
Chore: refactor code to better fit OOP

Agentum07 committed:
Chore: fix bot functionality, improved output

Agentum07 committed:
Chore: updating and refactoring code

Agentum07 committed:
Chore: add descriptive comments

Agentum07 committed:
Feat: create a state object for undo functionality

Agentum07 committed:
Chore: fix null pointer error

Agentum07 committed:
Remove DukeConstants.java to improve functionality

Agentum07 committed:
Change method parameters, directly passing state now

Agentum07 committed:
Fixed Undo implementation - can undo all tasks

Agentum07 committed:
Add javadocs for all public methods

Agentum07 committed:
Add greeting message

Agentum07 committed:
Add greeting message

Agentum07 committed:
Add background, change gui look

Agentum07 committed:
Merge pull request #5 from Agentum07/branch-A-BetterGui

Add background, change gui look

Agentum07 committed:
Delete dead import statements

Agentum07 committed:
Implement Undo done functionality

Agentum07 committed:
Refactor to fit coding standards

Agentum07 committed:
Fix checkstyle

Agentum07 committed:
Fix Gui

Agentum07 committed:
Update README.md

Agentum07 committed:
Add files via upload

Agentum07 committed:
Chore: delete dead code

Agentum07 committed:
Merge branch 'master' of https://github.com/Agentum07/ip

Agentum07 committed:
Fix code quality

Agentum07 committed:
Merge pull request #6 from Agentum07/branch-A-CodeQuality

Improve code quality

Agentum07 committed:
Add files via upload

Agentum07 committed:
Delete Ui.PNG

Agentum07 committed:
fix done command issue
